.github/workflows/docker-and-k8s.yml

```diff
@@ -0,0 +1,110 @@
+name: Build agent packages.
+
+on:
+  workflow_call:
+    inputs:
+      image-type:
+        description: "Name of the image build type."
+        required: true
+        type: string
+      python-version:
+        required: false
+        type: string
+        default: 3.8.10
+      publish:
+        description: "If true, then the result image will be published on master branch or any tag."
+        type: boolean
+        required: false
+        default: true
+
+    secrets:
+      SCALYR_PROD_CLOUDTECH_TESTING_WRITE_TOKEN:
+        required: true
+      SCALYR_CLOUDTECH_TESTING_DEV_SCALYR_READ_API_KEY:
+        required: true
+
+env:
+  DOCKER_BUILDKIT: 1
+
+
+jobs:
+  # TODO move here tests from the CicleCi.
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install python.
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ inputs.python-version }}
+
+      - name: Perform the deployment of the test environment.
+        uses: ./.github/actions/perform-deployment
+        with:
+          deployment-name: "test_environment"
+
+      - name: Do internal tools tests.
+        run: |
+          python3 -m pytest agent_build/tools/tests
+
+      - name: Set up QEMU
+        id: qemu
+        uses: docker/setup-qemu-action@v1
+        with:
+          image: tonistiigi/binfmt:qemu-v6.1.0
+          platforms: all
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+
+      - name: Start minikube for the test of the kubernetes build.
+        if: inputs.image-type == 'k8s'
+        shell: bash
+        run: |
+          minikube start
+
+      - name: Run image test.
+        env:
+            SCALYR_API_KEY: ${{ secrets.SCALYR_PROD_CLOUDTECH_TESTING_WRITE_TOKEN }}
+        run: |
+          python3 tests/package_tests/run_package_test.py run-package-test \
+            ${{ inputs.image-type}}_test \
+            --name-suffix "_${{ github.run_number }}${{ github.run_attempt }}"
+
+
+  publish:
+    needs: [test]
```

Comment:
Is there any way for us to run this workflow only after Circle CI tests pass? Or that would be too hard to do?

Reply:
Not so easy, I'm afrad.